### PR TITLE
main: Disconnect monitor change signals when window is destroyed

### PR DESCRIPTION
**Pull request by smspillaz**
Otherwise we'll attempt to mutate this._window

https://phabricator.endlessm.com/T18975

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1718,6 +1718,9 @@ const DiscoveryFeedApplication = new Lang.Class({
         // to it so that we can re-create it later
         this._window.connect('destroy', Lang.bind(this, function() {
             this._window = null;
+            this._discoveryFeedCardModel = new Gio.ListStore({
+                item_type: Stores.DiscoveryFeedCardStore.$gtype
+            });
```
Comment (cosimoc) [inline src/main.js]: Instead of doing this, you could simply create the model together with the window, no?

Comment (smspillaz) [reply src/main.js]: Yeah we could do that. My thought here was that the model was a property of the application and not the window, but the window is the only thing it is really used with...

**Comment by cosimoc** [comment]
@smspillaz Needs rebase to master

**Comment by smspillaz** [comment]
Rebased.

**Pull request merged**